hs/WebParsing/TimeTableParser.hs

```diff
@@ -17,15 +18,84 @@ import WebParsing.TimeConverter
 import Control.Monad.IO.Class (liftIO, MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Config (databasePath, timetableUrl)
+import Database.Tables (Department(Department))
 
 parseTT :: IO ()
 parseTT = do
     rsp <- simpleHTTP (getRequest timetableUrl)
     body <- getResponseBody rsp
+
     let depts = getDeptList $ parseTags body
+        deptCode = getDeptCodes $ parseTags body
+        deptName = getDeptName $ parseTags body
+        deptAssocList = init $ constructAssociationList deptName deptCode
 
     runSqlite databasePath $
         mapM_ getDeptTimetable depts
+    runSqlite databasePath $ 
+        mapM_ insertDepartment deptAssocList
+
+-- | List of unwanted words to remove when getting department codes.
+wordsToRemove :: [String]
+wordsToRemove = ["courses",
+                "courses]", 
+                "includes",
+                "and",
+                "-",
+                "[",
+                "]",
+                ""]
+
+-- | Extracts a list of lists of department codes from the main website
+getDeptCodes :: [Tag String] -> [[T.Text]]
```

Comment:
@alexbaluta A good way to simplify this code is to break up the parsing of Tags with the parsing of strings.

I want to see one function with the following type, which does the string parsing:

```haskell
parseLinkText :: String -> ([T.Text], -- the department codes
                            [T.Text] -- the department)
```

This has the advantage of being very easy to test, because you can just call it directly in the interpreter.